test/common/shared_functions.go

```diff
@@ -217,3 +220,65 @@ func WriteRHMICRToFile(client dynclient.Client, file string) error {
 		return writeObjToYAMLFile(rhmi, file)
 	}
 }
+
+// Common function to perform CRUDL and verifying their expected permissions
+func verifyCRUDLPermissions(t *testing.T, openshiftClient *resources.OpenshiftClient, expectedPermission ExpectedPermissions) {
+	// Perform LIST Request
+	resp, err := openshiftClient.DoOpenshiftGetRequest(expectedPermission.ListPath)
+
+	if err != nil {
+		t.Errorf("failed to perform LIST request with error : %s", err)
+	}
+
+	if resp.StatusCode != expectedPermission.ExpectedListStatusCode {
+		t.Errorf("unexpected response from LIST request : %v", resp)
```

Comment:
We could also print the expected response which would be nice for debugging

Reply:
Yup, sounds good to me, will update error messages to include the expected status code :+1: